esss_fix_format/cli.py

```diff
@@ -32,13 +32,25 @@
 def is_cpp(filename):
     """Return True if the filename is of a type that should be treated as C++ source."""
     from fnmatch import fnmatch
-    return any(fnmatch(os.path.split(filename)[-1], p) for p in CPP_PATTERNS)
+    return any(fnmatch(os.path.basename(filename), p) for p in CPP_PATTERNS)
 
 
 def should_format(filename):
-    """Return True if the filename is of a type that is supported by this tool."""
+    """
+    Return a tuple (fmt, reason) where fmt is True if the filename
+    is of a type that is supported by this tool.
+    """
     from fnmatch import fnmatch
-    return any(fnmatch(os.path.split(filename)[-1], p) for p in PATTERNS)
+    filename_no_ext, ext = os.path.splitext(filename)
+    ipynb_filename = filename_no_ext + '.ipynb'
+    # ignore .py file that has a jupytext configured notebook with the same base name
+    if ext == '.py' and os.path.isfile(ipynb_filename):
+        with open(ipynb_filename) as f:
+            if 'jupytext' in f.read():
```

Comment:
I did a search for `jupytext` in https://github.com/ESSS/notebooks and got no hits.
Is in new versions this string found?

Reply:
We are not using Jupytext in https://github.com/ESSS/notebooks.

As documented in `https://github.com/mwouts/jupytext`, if you want that jupyter notebook auto-saves python files when you edit the notebook (and vice-versa), you need to add the following line into the notebook metadata.
```
"jupytext": {"formats": "ipynb,py"},
```
So that's why I search for `jupytext`.
fix format ignores notebooks by default, but these `.py` auto generated by `Jupytext` needs to be ignored too.

Reply:
Thank you =)
I didn't know that.